### DISCUSSION TITLE
How to reuse Router?

### DISCUSSION THREAD

**Discussion by 2snEM6** [salvo-rs/salvo#100] [Q&A] [answered]
Hi! I am trying to reuse the same router, since /v1 and /latest need to point to the v1 router.

<img width="655" alt="image" src="https://user-images.githubusercontent.com/7485885/179356815-bdd8601d-9a17-49c2-a53e-12dcc21943fd.png">

However, I am getting the typical `valued moved` error.

Is there any way around this?

Thanks!

**Comment by chrislearn** [comment, marked as answer]
You can use function to create the router and call it twice. 
Like this:

```rust
fn v1_router()  -> Router {
  Router::new().
  ....
}
```

**Reply by 2snEM6** [reply]
Just did that! Thanks!
